maven-plugin/src/main/java/org/infinispan/plugins/proto/compatibility/ProtoCompatibilityMojo.java

```diff
@@ -92,4 +91,32 @@ public void execute() throws MojoExecutionException {
          throw new MojoExecutionException("An error occurred while running protolock", e);
       }
    }
+
+   private ProtoLock protoLockFromDir(Path protoRoot) throws IOException {
+      List<Path> protoFiles;
+      try (Stream<Path> stream = Files.walk(protoRoot)) {
+         protoFiles = stream.filter(p -> p.getFileName().toString().endsWith(".proto")).toList();
+      }
+      FileDescriptorSource fds = new FileDescriptorSource();
+      for (Path p : protoFiles) {
+         fds.addProtoFile(protoRoot.relativize(p).toString(), p.toFile());
+      }
+      ProtostreamProtoParser parser = new ProtostreamProtoParser(Configuration.builder().build());
+      Map<String, FileDescriptor> descriptors = parser.parse(fds);
+      return new ProtoLock(descriptors.values());
+   }
+
+   private void checkRemoteCompatibility(ProtoLock currentState) throws IOException {
+      if (remoteLockFiles.isEmpty())
+         return;
+
+      for (String file : remoteLockFiles.split(",")) {
+         getLog().info(String.format("Checking backwards compatibility check against remote file '%s'", file));
+         try (InputStream is = new URL(file).openStream()) {
```

Comment:
In offline mode we may want to allow URLs with the `file` protocol

Reply:
We already have `protoSourceRoot` and `protoLockRoot` for defining checks against local files.